docs/rackhd/tftp_dhcp_server.rst

```diff
@@ -0,0 +1,313 @@
++===========================+=============================================+=================================================+
+| TFTP and DHCP service are | 1. on-tftp(default)                         | 1. ISC DHCP + on-dhcp-proxy(default)            |
+| in RackHD server          | 2. Third-party TFTP service such as         | 2. ISC DHCP only                                |
+|                           |    in.tftpd(tftp-hpa) in Ubuntu OS          | 3. Third-party DHCP Service + DHCP proxy        |
+|                           |                                             | 4. Third-party DHCP Service only                |
++---------------------------+---------------------------------------------+-------------------------------------------------+
+| TFTP and DHCP service are | 1. on-tftp                                  | 1. ISC DHCP + on-dhcp-proxy                     |
+| in separate server        | 2. Third-party TFTP service such as         | 2. ISC DHCP only                                |
+|                           |    in.tftpd(tftp-hpa) in Ubuntu OS          | 3. Third-party DHCP Service + DHCP proxy        |
+|                           |                                             | 4. Third-party DHCP Service only                |
++---------------------------+---------------------------------------------+-------------------------------------------------+
+
+**NOTE:** "Third-party" service means it's not RackHD default service.
+
+TFTP and DHCP are in RackHD Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TFTP Service Configuration in RackHD Server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Default on-tftp Configuration
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+RackHD default TFTP service is `on-tftp`_, it could be configured by fields `tftpBindAddress`, `tftpBindPort`, `tftpRoot`
+in `config.json`_, and `RackHD iPXE`_ files are put to `tftpRoot` directory.
+
+.. code-block:: javascript
+
+    ...
+    "tftpBindAddress": "172.31.128.1",
+    "tftpBindPort": 69,
+    "tftpRoot": "./static/tftp",
+    ...
+
+Third-Party TFTP Service Configuration
+""""""""""""""""""""""""""""""""""""""
+
+.. _RackHD TFTP Templates: https://github.com/RackHD/on-tftp/tree/master/data/templates
+
+Any other TFTP service could be used by RackHD, just put `RackHD iPXE`_ files to TFTP root directory.
+For scripts in `RackHD TFTP Templates`_, if the parameters such as `apiServerAddress`, `apiServerPort` are rendered by `on-tftp`,
+they need to be hardcoded, it's `172.31.128.1` and `9080` in the example. then put these scripts to TFTP root directory.
+
+**NOTE**:
+  1. If all managed nodes' NIC ROM is iPXE, not PXE, don't need to put `RackHD iPXE`_ files to TFTP directory.
+  2. If the funtionality supported by rendered scripts are not needed, don't need to put `RackHD TFTP Templates`_ scripts to TFTP directory.
```

Comment:
"If the funtionality supported by rendered scripts is not needed, then you don't need to put `RackHD TFTP Templates`_ scripts to TFTP directory."

(the singular is/are is from "functionality", not the plural "rendered scripts", in this sentence)